backend/app/src/test/kotlin/io/tolgee/api/v2/controllers/v2ProjectsController/ProjectsControllerEditTest.kt

```diff
@@ -72,4 +74,36 @@ class ProjectsControllerEditTest : AuthorizedControllerTest() {
       node("CUSTOM_VALIDATION.namespaces_cannot_be_disabled_when_namespace_exists").isNotNull
     }
   }
+
+  @Test
+  fun `can use the same slug as deleted project has`() {
+    val base = dbPopulator.createBase()
+
+    executeInNewTransaction {
+      val deleted = dbPopulator.createBase()
+      deleted.project.deletedAt = currentDateProvider.date
+      deleted.project.slug = "new-slug-2"
+      projectService.save(deleted.project)
+    }
+    val newSlug = "new-slug-2"
+    val content =
+      EditProjectRequest(
+        name = "new name",
+        baseLanguageId = base.project.languages.toList()[1].id,
+        slug = "new-slug-2",
```

Comment:
```suggestion
        slug = newSlug,
```

Or `slugToBeReused` if you find the previous comment useful.